robottelo/enums.py

```diff
@@ -0,0 +1,69 @@
+"""Module containing enumeration classes used throughout Robottelo.
+
+This module provides standardized enumerations for various types, statuses,
+and configurations used in Robottelo tests and utilities.
+"""
+
+from enum import StrEnum
+
+
+class NetworkType(StrEnum):
+    """
+    Enumeration of host network addressing types.
+
+    This enum represents the different network addressing modes that can be
+    configured for a host.
+    """
+
+    IPV4 = 'ipv4'
+    IPV6 = 'ipv6'
+    DUALSTACK = 'dualstack'
+
+    @property
+    def formatted(self):
+        """
+        Returns the properly formatted version of the network type (e.g., 'IPv4', 'IPv6')
+
+        :returns: The formatted network type as a string.
+        :rtype: str
+        :raises ValueError: If called on DUALSTACK
+        """
+        if self.value == 'dualstack':
+            raise ValueError(f'Formatted property not supported for {self.name}')
+        if self.value in (self.IPV4, self.IPV6):
+            return self.value.replace('ipv', 'IPv')
+        raise ValueError(f'Invalid network type: {self.value}')
+
+    def __str__(self):
+        """
+        Returns the string representation of the enum member.
+
+        :returns: The string value of the enum member.
+        :rtype: str
+        """
+        return self.value
+
+    def __eq__(self, other):
```

Comment:
This duplicates the built in implementation of `StrEnum`:
```python
from enum import StrEnum

class NetworkType(StrEnum):
    IPV4 = 'ipv4'
    IPV6 = 'ipv6'
    DUALSTACK = 'dualstack'

print(NetworkType.IPV4 == 'ipv4')
```

This prints True for me

Reply:
Good catch, the enum was initially `Enum`, not `StrEnum` as it is now. I will remove the `__eq__` method.